### PR TITLE
Validate that a docker image referenced in WDL payload validates

### PR DESCRIPTION
**Pull request by gbggrant**
Validate that a docker image referenced in WDL payload validates (that it exists in docker hub).

Validate that if WDL contains a docker reference and it is properly formatted,
that the docker image exists.  Dealt with both 'official' and 'personal'
docker hub images and optional tags.  Added integration tests.

### CODE REVIEW

**Comment by geoffjentry** [comment]
Note that at least for now JES requires docker images to be on gcr

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/util/DockerHubClient.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  import actorSystem.dispatcher
+
+  val timeout = 5.seconds
+
+  def pipeline = sendReceive
+
+  def gcsGET(url: String): Future[HttpResponse] = pipeline(Get(url))
+
+  def doesDockerImageExist(dockerImage: DockerImageReference): Boolean = {
+    val dockerImageInfo = gcsGET(dockerImageTagUrl(dockerImage))
```
Comment (jacarey) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/util/DockerHubClient.scala]: gcs?

**Comment by jacarey** [comment]
:+1:

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/util/DockerHubClient.scala (hunk 1 above)
Comment (gbggrant) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/util/DockerHubClient.scala]: Renamed...

**Pull request merged**